tutorials/quick_start/pytorch_fw/ultralytics_lib/replacers.py

```diff
@@ -304,25 +115,45 @@ def val(self, data=None, **kwargs):
             args.imgsz = self.model.args['imgsz']  # use trained imgsz unless custom value is passed
         args.imgsz = check_imgsz(args.imgsz, max_dim=1)
 
-        validator = TASK_MAP[self.task][2](args=args)
+        validator = TASK_MAP[self.task]['validator'](args=args)
+        validator.model = self.model
         validator(model=self.model)
         self.metrics_data = validator.metrics
 
         return validator.metrics
 
 
-def prepare_model_for_ultralytics_val(ultralytics_model, model):
+def prepare_model_for_ultralytics_val(ultralytics_model, model, quantized=False):
     """
     Prepares the model for Ultralytics validation by setting necessary attributes.
     """
-    if not hasattr(model, 'args'):
-        def fuse():
-            return ultralytics_model.model
+    class CustomPytorchModel(PytorchModel):
```

Comment:
I would define the class outside the function. Also add a description so we will be able to understand why it's needed.

Reply:
I actually think it fits better inside since it is only relevant for this fix